### PR TITLE
Issue 4051: Create pravega build with jdk11 docker image

### LINKED ISSUE #4051: Create Pravega builds with jdk11  docker image
**Problem description**

Currently, pravega builds are using jdk8 docker image.  Create one more pravega build, that uses jdk11 docker image.

**Problem location**

**Suggestions for an improvement**
Maintain 2 docker files, one for jdk8 and other for jdk11

### PR DESCRIPTION
**Pull request by anishakj**
**Change log description**  
* Creates a docker file based on `openjdk:11.0-jdk-slim`.
* Uses the new docker file if the build is triggered using `pravega-java-11`.

**Purpose of the change**  
Fixes #4051 

**What the code does**  
 Created separate docker file with jdk11 image. Based on the java version, different docker files will be used for docker build. Also changed `sh` shell to `bash` , because `openjdk:11.0-jdk-slim` is using the default shell `dash`

Pravega docker image size with ` openjdk:11.0-jdk-slim` is around `569 MB` and with `8u181-jre-alpine` is around `250 MB`. Also, `openjdk11` does not support `alpine` based images ref: https://github.com/docker-library/openjdk/issues/313

**How to verify it**  
Verify system tests are passing

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -842,6 +842,9 @@ project('test:system') {
         // tier2 Configuration, specified as comma seperated key values k1=v1,k2=v2
         systemProperty "tier2Config", System.getProperty("tier2Config")
 
+        //testPodImage , default is openjdk:8u181-jre-alpine
+        systemProperty "testPodImage", System.getProperty("testPodImage", "openjdk:8u181-jre-alpine")
```
Comment (shrids) [inline build.gradle]: This is related to issue #4040 right?

File: build.gradle [hunk 2]
```diff
@@ -1060,11 +1063,17 @@ task preparePravegaImage(type: Copy) {
 task buildPravegaImage(type: DockerBuildTask, dependsOn: preparePravegaImage) {
     baseTag = pravegaBaseTag
     dockerDir = preparePravegaImage.destinationDir.absolutePath
+    if (JavaVersion.current() == JavaVersion.VERSION_1_8) {
```
Comment (shrids) [inline build.gradle]: We could use the following approach 
```
if (JavaVersion.isJava11Compatible()) { // returns true if the current java version is >= Java 11.
    dockerFile = "${dockerDir}/Dockerfile-java-11"
} else {
   dockerFIle = "${dockerDir}/Dockerfile"
}
```
or the other options is to have two tasks for java 8 based docker image and one for java 11 based docker image.

File: docker/pravega/Dockerfile-java-11 [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    iproute2  \
+  && rm -rf /var/lib/apt/lists/*
+
+EXPOSE 9090 9091 10000 12345
+
+WORKDIR /opt/pravega
+
+COPY pravega/ /opt/pravega/
+COPY scripts/ /opt/pravega/scripts/
+RUN chmod +x -R /opt/pravega/scripts/ && chsh -s /bin/bash
```
Comment (shrids) [inline docker/pravega/Dockerfile-java-11]: could we add a comment explaining why `chsh` is being used? i.e. explain about the default shell and talk about the health check commands executed by the operator.

File: build.gradle (hunk 1 above)
Comment (anishakj) [reply build.gradle]: yes, the fix is merged only in `r0.5`

File: build.gradle (hunk 2 above)
Comment (anishakj) [reply build.gradle]: I  have made the changes.

File: docker/pravega/Dockerfile-java-11 (hunk 3 above)
Comment (anishakj) [reply docker/pravega/Dockerfile-java-11]: Done

File: docker/pravega/Dockerfile-java-11 [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+EXPOSE 9090 9091 10000 12345
+
+WORKDIR /opt/pravega
+
+COPY pravega/ /opt/pravega/
+COPY scripts/ /opt/pravega/scripts/
+
+# Default shell of jdk11 image is dash.
+# If operator changes health check command,
```
Comment (shrids) [inline docker/pravega/Dockerfile-java-11]: could we rephrase this ?

File: docker/pravega/Dockerfile-java-11 [hunk 5]
```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+FROM openjdk:11.0-jdk-slim
```
Comment (shrids) [inline docker/pravega/Dockerfile-java-11]: Observing that `openjdk:11.0-jre-slim` has a smaller size(`204MB`) . We should try this docker image and update the findings.

Comment (anishakj) [reply docker/pravega/Dockerfile-java-11]: Changed the image to `openjdk:11.0-jre-slim`.

File: docker/pravega/Dockerfile-java-11 [hunk 6]
```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (fpj) [inline docker/pravega/Dockerfile-java-11]: Please remove year from copyright statement. It should be:

```
Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
```

File: build.gradle [hunk 7]
```diff
@@ -1060,11 +1063,17 @@ task preparePravegaImage(type: Copy) {
 task buildPravegaImage(type: DockerBuildTask, dependsOn: preparePravegaImage) {
     baseTag = pravegaBaseTag
     dockerDir = preparePravegaImage.destinationDir.absolutePath
+    if (JavaVersion.current().isJava11Compatible()) {
```
Comment (fpj) [inline build.gradle]: Where do I specify the Java version?

Comment (anishakj) [reply build.gradle]: we are not specifying the Java version. It looks for the current Java version installed and based on the version select the docker files.
For pravega-java-11 job, we are setting the Java path to JDK11

File: docker/pravega/Dockerfile-java-11 (hunk 6 above)
Comment (anishakj) [reply docker/pravega/Dockerfile-java-11]: Done

**Pull request merged**